src/fu-util.c

```diff
@@ -2365,11 +2365,169 @@ fu_util_modify_config (FuUtilPrivate *priv, gchar **values, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_util_upload_security (FuUtilPrivate *priv,
+			 GPtrArray *attrs,
+			 FwupdRemote *remote,
+			 GError **error)
+{
+	guint status_code;
+	GHashTableIter iter;
+	const gchar *key;
+	const gchar *value;
+	g_autofree gchar *data = NULL;
+	g_autofree gchar *sig = NULL;
+	g_autoptr(GHashTable) metadata = NULL;
+	g_autoptr(JsonBuilder) builder = NULL;
+	g_autoptr(JsonGenerator) json_generator = NULL;
+	g_autoptr(JsonNode) json_root = NULL;
+	g_autoptr(SoupMessage) msg = NULL;
+	g_autoptr(SoupMultipart) mp = NULL;
+
+	/* set up networking */
+	if (priv->soup_session == NULL) {
+		priv->soup_session = fu_util_setup_networking (error);
+		if (priv->soup_session == NULL)
+			return FALSE;
+	}
+
+	/* get metadata */
+	metadata = fwupd_client_get_report_metadata (priv->client,
+						     priv->cancellable,
+						     error);
+	if (metadata == NULL)
+		return FALSE;
+
+	/* create header */
+	builder = json_builder_new ();
+	json_builder_begin_object (builder);
+	json_builder_set_member_name (builder, "ReportVersion");
+	json_builder_add_int_value (builder, 2);
+	json_builder_set_member_name (builder, "MachineId");
+	json_builder_add_string_value (builder, fwupd_client_get_host_machine_id (priv->client));
+
+	/* this is system metadata not stored in the database */
+	json_builder_set_member_name (builder, "Metadata");
+	json_builder_begin_object (builder);
+
+	g_hash_table_iter_init (&iter, metadata);
+	while (g_hash_table_iter_next (&iter, (gpointer *) &key, (gpointer *) &value)) {
+		json_builder_set_member_name (builder, key);
+		json_builder_add_string_value (builder, value);
+	}
+	json_builder_set_member_name (builder, "HostSecurityId");
+	json_builder_add_string_value (builder, fwupd_client_get_host_security_id (priv->client));
+	json_builder_end_object (builder);
+
+	/* attrs */
+	json_builder_set_member_name (builder, "SecurityAttributes");
+	json_builder_begin_array (builder);
+	for (guint i = 0; i < attrs->len; i++) {
+		FwupdSecurityAttr *attr = g_ptr_array_index (attrs, i);
+		json_builder_begin_object (builder);
+		fwupd_security_attr_to_json (attr, builder);
+		json_builder_end_object (builder);
+	}
+	json_builder_end_array (builder);
+	json_builder_end_object (builder);
+
+	/* export as a string */
+	json_root = json_builder_get_root (builder);
+	json_generator = json_generator_new ();
+	json_generator_set_pretty (json_generator, TRUE);
+	json_generator_set_root (json_generator, json_root);
+	data = json_generator_to_data (json_generator, NULL);
+	if (data == NULL) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_INTERNAL,
+				     "Failed to convert to JSON string");
+		return FALSE;
+	}
+
+	/* self sign data */
+	if (priv->sign) {
+		sig = fwupd_client_self_sign (priv->client, data,
+					      FWUPD_SELF_SIGN_FLAG_ADD_TIMESTAMP,
+					      priv->cancellable, error);
+		if (sig == NULL)
+			return FALSE;
+	}
+
+	/* ask for permission */
+	if (!priv->assume_yes) {
```

Comment:
aren't you missing the stuff that lets this go automatic here?

Reply:
I am! Fixed.